conans/client/tools/files.py

```diff
@@ -101,41 +101,42 @@ def print_progress(the_size, uncomp_size):
         def print_progress(_, __):
             pass
 
-    with zipfile.ZipFile(filename, "r") as z:
-        if not pattern:
-            zip_info = z.infolist()
-        else:
-            zip_info = [zi for zi in z.infolist() if fnmatch(zi.filename, pattern)]
-        uncompress_size = sum((file_.file_size for file_ in zip_info))
-        if uncompress_size > 100000:
-            output.info("Unzipping %s, this can take a while" % human_size(uncompress_size))
-        else:
-            output.info("Unzipping %s" % human_size(uncompress_size))
-        extracted_size = 0
-
-        print_progress.last_size = -1
-        if platform.system() == "Windows":
-            for file_ in zip_info:
-                extracted_size += file_.file_size
-                print_progress(extracted_size, uncompress_size)
-                try:
-                    z.extract(file_, full_path)
-                except Exception as e:
-                    output.error("Error extract %s\n%s" % (file_.filename, str(e)))
-        else:  # duplicated for, to avoid a platform check for each zipped file
-            for file_ in zip_info:
-                extracted_size += file_.file_size
-                print_progress(extracted_size, uncompress_size)
-                try:
-                    z.extract(file_, full_path)
-                    if keep_permissions:
-                        # Could be dangerous if the ZIP has been created in a non nix system
-                        # https://bugs.python.org/issue15795
-                        perm = file_.external_attr >> 16 & 0xFFF
-                        os.chmod(os.path.join(full_path, file_.filename), perm)
-                except Exception as e:
-                    output.error("Error extract %s\n%s" % (file_.filename, str(e)))
-        output.writeln("")
+    if zipfile.is_zipfile(filename):
```

Comment:
I don't understand this, this code is specifically for zip files, so no need to check it. 
At most it could be a "precondition" check that checks it and raise a meaningful error otherwise, but that would be a different feature (that will require its own unit test)

Reply:
When I use to run the command `conan configure install` on a file, the file is categorized as a file, then, try to unzip this file and this raise an exception, because it's not a ZIP file.
My meaning, in this change, is to check if the file is a ZIP file, first, to avoid unnecessary exception and after that threat the file as a real file.
I can change that in order to first check if the input file is a configuration, and then, run the unzip function in order to keep the raising error. Or I can keep this change and add a meaning full error output. What is your preference ?